packages/mui-base/src/useSelect/useSelect.ts

```diff
@@ -351,24 +370,42 @@ function useSelect<OptionValue, Multiple extends boolean = false>(
     externalProps: ExternalProps = {} as ExternalProps,
   ): UseSelectButtonSlotProps<ExternalProps> => {
     const externalEventHandlers = extractEventHandlers(externalProps);
-    const listboxAndButtonProps = combineHooksSlotProps(getButtonRootProps, getListboxRootProps);
-    const combinedProps = combineHooksSlotProps(listboxAndButtonProps, getSelectTriggerProps);
+    const propGetter =
+      focusManagement === 'activeDescendant'
+        ? combineHooksSlotProps(getButtonRootProps, getListboxRootProps)
+        : getButtonRootProps;
+
+    const combinedProps = combineHooksSlotProps(propGetter, getSelectTriggerProps);
+
     return {
```

Comment:
I have this type error here:

<img width="835" alt="Screenshot 2023-10-30 at 13 04 45" src="https://github.com/mui/material-ui/assets/16889233/d2253ee3-f77a-442f-b5ed-618713345e7f">
<br></br>

It's caused by the `tabIndex` property being `number` in `UseListRootSlotProps`. I need to make it `number | undefined` as it won't come for DOM focus management. I thought that [`Omit<UseListRootSlotProps, 'tabIndex'>` here](https://github.com/mui/material-ui/pull/39675/files#diff-fb83492cad9b8993a4f40f8361d749c4bd03620c45edf0c3d530f3d6bf9c4b6aR138-R140) would do the trick. And it does, but something is going on with `TOther` that breaks it. I couldn't figure it out.